scanpy/tests/test_clustering.py

```diff
@@ -13,10 +14,75 @@ def adata_neighbors():
 
 
 @needs.leidenalg
-def test_leiden_basic(adata_neighbors):
-    sc.tl.leiden(adata_neighbors)
+@needs.igraph
+@pytest.mark.parametrize("flavor", ["igraph", "leidenalg", "leidenalg"])
```

Comment:
Why is `leidenalg` duplicated here? also, you had `leidenalg` twice

```suggestion
@pytest.mark.parametrize(
    "flavor",
    [
        pytest.param("igraph", marks=needs.igraph),
        pytest.param("leidenalg", marks=needs.leidenalg),
    ],
)
```